.evergreen/config.yml

```diff
@@ -1,1303 +1,86 @@
-          SKIP_CRYPT_SHARED=${SKIP_CRYPT_SHARED} \
-          SSL=${SSL} \
-          SSL_DIR=${SSL_DIR} \
-          TESTS=${TESTS} \
-          sh ${PROJECT_DIRECTORY}/.evergreen/run-tests.sh
-
-  "cleanup":
-    - command: shell.exec
-      params:
-        script: |
-          ${PREPARE_SHELL}
-          rm -rf $DRIVERS_TOOLS || true
-
-  "fix absolute paths":
-    - command: shell.exec
-      params:
-        script: |
-          ${PREPARE_SHELL}
-          for filename in $(find ${DRIVERS_TOOLS} -name \*.json); do
-            perl -p -i -e "s|ABSOLUTE_PATH_REPLACEMENT_TOKEN|${DRIVERS_TOOLS}|g" $filename
-          done
-
-  "windows fix":
-    - command: shell.exec
-      params:
-        script: |
-          ${PREPARE_SHELL}
-          for i in $(find ${DRIVERS_TOOLS}/.evergreen ${PROJECT_DIRECTORY}/.evergreen -name \*.sh); do
-            cat $i | tr -d '\r' > $i.new
-            mv $i.new $i
-          done
-
-  "make files executable":
-    - command: shell.exec
-      params:
-        script: |
-          ${PREPARE_SHELL}
-          for i in $(find ${DRIVERS_TOOLS}/.evergreen ${PROJECT_DIRECTORY}/.evergreen -name \*.sh); do
-            chmod +x $i
-          done
-
-  "init test-results":
-    - command: shell.exec
-      params:
-        script: |
-          ${PREPARE_SHELL}
-          echo '{"results": [{ "status": "FAIL", "test_file": "Build", "log_raw": "No test-results.json found was created"  } ]}' > ${PROJECT_DIRECTORY}/test-results.json
-
-  "install dependencies":
-    - command: shell.exec
-      params:
-        working_dir: "src"
-        script: |
-          ${PREPARE_SHELL}
-          file="${DRIVERS_TOOLS}/.evergreen/install-dependencies.sh"
-          # Don't use ${file} syntax here because evergreen treats it as an empty expansion.
-          [ -f "$file" ] && sh $file || echo "$file not available, skipping"
-
-  "decrypt atlas URI list":
-    - command: shell.exec
-      params:
-        script: |
-          openssl aes-256-cbc -S "${encrypted_uris_salt}" -K "${encrypted_uris_key}" -iv "${encrypted_uris_iv}" -in ${PROJECT_DIRECTORY}/.evergreen/atlas-uris.txt.enc -out ${PROJECT_DIRECTORY}/.evergreen/atlas-uris.txt -d
-
-  "start load balancer":
-    - command: shell.exec
-      params:
-        script: |
-          MONGODB_URI="${MONGODB_URI}" ${DRIVERS_TOOLS}/.evergreen/run-load-balancer.sh start
-    - command: expansions.update
-      params:
-        file: lb-expansion.yml
-
-  "stop load balancer":
-    - command: shell.exec
-      params:
-        script: |
-          # Only run if a load balancer was started
-          if [ -n "${SINGLE_MONGOS_LB_URI}" ]; then
-            ${DRIVERS_TOOLS}/.evergreen/run-load-balancer.sh stop
-          fi
-
-pre:
+# These pre and post rules apply to all tasks not part of a task group, which should only ever be tests against local
+# MongoDB instances. All other tasks that require special preparation should be run from within a task group
+pre: &test_setup
```

Comment:
What references `*test_setup` and `*test_teardown`?

Reply:
In this file, nothing. Off the top of my head I'm not sure whether these could be used in included files as well, but since they're not used in any case I've removed the tags.